src/only-export-components.ts

```diff
@@ -56,7 +59,7 @@ export const onlyExportComponents: TSESLint.RuleModule<
       checkJS = false,
       allowExportNames,
     } = context.options[0] ?? {};
-    const filename = context.getFilename();
+    const filename = context.filename;
```

Comment:
`context.getFilename()` is deprecated, see https://eslint.org/docs/latest/extend/custom-rules#the-context-object